src/components/capture/creation/SpecFormHeader.tsx

```diff
@@ -20,7 +20,7 @@ function NewCaptureSpecFormHeader() {
     const { data, error } = useConnectorImages(endpoint);
 
     useEffect(() => {
-        if (data?.links.spec && data.links.spec.length > 0) {
+        if (endpoint && data?.links.spec && data.links.spec.length > 0) {
```

Comment:
Need to make sure there is an endpoint before setting the spec as it depends on the endpoint